contribs/gmf/src/directives/objecteditinggetwmsfeature.js

```diff
@@ -0,0 +1,159 @@
+goog.provide('gmf.ObjecteditinggetwmsfeatureController');
+goog.provide('gmf.objecteditinggetwmsfeatureDirective');
+
+goog.require('gmf');
+goog.require('gmf.ObjectEditingQuery');
+
+
+/**
+ * When activated, this directive registers clicks on an OL3 map and use the
+ * clicked coordinate to fetch a feature using the ObjectEditing query service.
+ * A feature returned is pushed to a collection.
+ *
+ * Example:
+ *
+ *     <gmf-objecteditinggetwmsfeature
+ *         gmf-objecteditinggetwmsfeature-active="ctrl.active"
+ *         gmf-objecteditinggetwmsfeature-features="ctrl.features"
+ *         gmf-objecteditinggetwmsfeature-layerinfo="ctrl.layerInfo"
+ *         gmf-objecteditinggetwmsfeature-map="::ctrl.map">
+ *     </gmf-objecteditinggetwmsfeature>
+ *
+ * @htmlAttribute {boolean} gmf-objecteditinggetwmsfeature-active Whether the
+ *     directive is active or not.
+ * @htmlAttribute {ol.Collection} gmf-objecteditinggetwmsfeature-features
+ *     The collection of features where to add those created by this directive.
+ * @htmlAttribute {gmf.ObjectEditingQuery.QueryableLayerInfo} gmf-objecteditinggetwmsfeature-layerinfo Queryable layer info.
+ * @htmlAttribute {ol.Map} gmf-objecteditinggetwmsfeature-map The map.
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfObjecteditinggetwmsfeature
+ */
+gmf.objecteditinggetwmsfeatureDirective = function() {
+  return {
+    controller: 'GmfObjecteditinggetwmsfeatureController',
+    scope: {
+      'active': '=gmfObjecteditinggetwmsfeatureActive',
+      'features': '<gmfObjecteditinggetwmsfeatureFeatures',
+      'layerInfo': '=gmfObjecteditinggetwmsfeatureLayerinfo',
+      'map': '<gmfObjecteditinggetwmsfeatureMap'
+    },
+    bindToController: true,
+    controllerAs: 'gwfCtrl'
+  };
+};
+
+gmf.module.directive(
+  'gmfObjecteditinggetwmsfeature',
+  gmf.objecteditinggetwmsfeatureDirective);
+
+
+/**
+ * @param {!angular.Scope} $scope Scope.
+ * @param {gmf.ObjectEditingQuery} gmfObjectEditingQuery GMF ObjectEditing
+ *     query service.
+ * @constructor
+ * @ngInject
+ * @ngdoc controller
+ * @ngname GmfObjecteditinggetwmsfeatureController
+ */
+gmf.ObjecteditinggetwmsfeatureController = function($scope,
+    gmfObjectEditingQuery) {
+
+  // Scope properties
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.active;
+
+  $scope.$watch(
+    function() {
+      return this.active;
+    }.bind(this),
+    this.handleActiveChange_.bind(this)
```

Comment:
Mhhh I don't now if that can be a problem or not... look at the the doc: https://docs.angularjs.org/api/ng/type/$rootScope.Scope : 

````
After a watcher is registered with the scope, the listener fn is called asynchronously (via $evalAsync) to initialize the watcher. In rare cases, this is undesirable because the listener is called when the result of watchExpression didn't change. To detect this scenario within the listener fn, you can compare the newVal and oldVal. If these two values are identical (===) then the listener was called due to initialization.
````